arduino-ide-extension/src/node/arduino-daemon-impl.ts

```diff
@@ -149,34 +151,12 @@ export class ArduinoDaemonImpl
   }
 
   private async debugDaemon(): Promise<boolean> {
-    // Poor man's preferences on the backend. (https://github.com/arduino/arduino-ide/issues/1056#issuecomment-1153975064)
-    const configDirUri = await this.envVariablesServer.getConfigDirUri();
-    const configDirPath = FileUri.fsPath(configDirUri);
-    try {
-      const raw = await fs.readFile(join(configDirPath, 'settings.json'), {
-        encoding: 'utf8',
-      });
-      const json = this.tryParse(raw);
-      if (json) {
-        const value = json['arduino.cli.daemon.debug'];
-        return typeof value === 'boolean' && !!value;
-      }
-      return false;
-    } catch (error) {
-      if (ErrnoException.isENOENT(error)) {
-        return false;
-      }
-      throw error;
-    }
-  }
-
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private tryParse(raw: string): any | undefined {
-    try {
-      return JSON.parse(raw);
-    } catch {
-      return undefined;
+    const settings = await this.settingsReader.read();
+    if (settings) {
+      const value = settings['arduino.cli.daemon.debug'];
+      return typeof value === 'boolean' && !!value;
```

Comment:
The strict equality operator (`===`) returns `false` if the types of the two operands are different.

```suggestion
      return value === true
```